sources/Algo/include/Algo.h

```diff
@@ -393,5 +398,300 @@ class ControllerInterfaceDefinitionAlgorithm : public NodeAlgorithm {
+      LINE,      ///< Line type
+      TFO,       ///< Transfo type
+      SHUNT      ///< Shunt type
+    };
+
+    /**
+     * @brief Constructor
+     *
+     * @param macroid the macro connector id to use
+     * @param type type of the element to connect
+     * @param id the element id to connect
+     */
+    MacroConnection(const MacroId& macroid, const ElementType& type, const ElementId& id) : id(macroid), elementType(type), connectedElementId(id) {}
+
+    /**
+     * @brief Equality operator
+     * @param other the macro connection to compare to
+     * @returns true if they are equal, false if not
+     */
+    bool operator==(const MacroConnection& other) const;
+
+    /**
+     * @brief Non-equality operator
+     * @param other the macro connection to compare to
+     * @returns true if they are not equal, false if not
+     */
+    bool operator!=(const MacroConnection& other) const;
+
+    /**
+     * @brief Less than operator
+     * @param other the macro connection to compare to
+     * @returns true if inferior than other, false if not
+     */
+    bool operator<(const MacroConnection& other) const;
+
+    /**
+     * @brief Less or equal than operator
+     * @param other the macro connection to compare to
+     * @returns true if inferior or equal than other, false if not
+     */
+    bool operator<=(const MacroConnection& other) const;
+
+    /**
+     * @brief Greater than operator
+     * @param other the macro connection to compare to
+     * @returns true if superior than other, false if not
+     */
+    bool operator>(const MacroConnection& other) const;
+
+    /**
+     * @brief Greater or equal than operator
+     * @param other the macro connection to compare to
+     * @returns true if superior or equal than other, false if not
+     */
+    bool operator>=(const MacroConnection& other) const;
+
+    MacroId id;                    ///< Macro connector id
+    ElementType elementType;       ///< connected element type
+    ElementId connectedElementId;  ///< Element id connected throught the macro connection (can be node, line or tfo)
+  };
+
+  /**
+   * @brief Constructor
+   *
+   * @param idauto the dynamic model id
+   * @param libauto the library name
+   */
+  DynModelDefinition(const DynModelId& idauto, const std::string& libauto) : id(idauto), lib(libauto) {}
+
+  DynModelId id;                              ///< dynamic model id
+  std::string lib;                            ///< library name
+  std::set<MacroConnection> nodeConnections;  ///< set of macro connections for the dynamic model
+};
+
+/**
+ * @brief Definition of models
+ */
+struct DynModelDefinitions {
+  std::unordered_map<DynModelDefinition::DynModelId, DynModelDefinition> models;          ///< models by dynamic model id
+  std::unordered_set<DynModelDefinition::MacroConnection::MacroId> usedMacroConnections;  ///< list of macro connectors used for current set of models
+};
+
+/**
+ * @brief DynModel algorithm
```

Comment:
Please be more specific in the comment

Reply:
The description of the algorithm, to be compliant with other algorithms, is described in the operator() comments. The main comment of the class is changed to "Algorithm to find dynamic models"